### PR TITLE
Update readme files for default resource directory

### LINKED ISSUE #860: Folder default/readme.txt needs updating
> buildings.txt, fields.txt, ship_hulls.txt, ship_parts.txt, specials.txt, species.txt, techs.txt: These define the basic content types referred to in their names. 

Many of these files either no longer exist or have been substantially reduced, the subfolder system needs explaining.

> Note that species.txt often contains the effects of techs/buildings, for technical reasons.

All of this has either been moved to a .macro file or put into the tech or building location.

### PR DESCRIPTION
**Pull request by dbenage-cx**
Fixes #860.

Updates `default/README.txt` to briefly describe each sub-directory and file.
Moves any sub-directory file descriptions to the relevant readme file.
To avoid duplicated work, assumes fonts will move to `default/data/fonts/` per #843, as it appears ready to merge.

Clarification or expanded summaries of some entries may be needed:
- customizations
- python/handlers

### CODE REVIEW

**Comment by adrianbroher** [comment]
@MatGB 

I'm not too sure if I want to agree to your argument.  Because:
1. The main project README.md is already Markdown, is delivered to the end user as documentation entry point and we already assume with that the user is able to handle the `open file with` route.
2. If the user is unable to handle the `open file with` route he probably has no clue anyway and should abstain from modifying anything.

@dbenage-cx 

Looks good so far.
- Please wrap lines over 80 chars for easier diffing.
- Add an empty line after titles.
- Is there a particular reason why you added spaces before and after the asterisk in the _filename guidelines_ section of the `default/scripting/README.md` file?
- If things are **Required** in the _Contents_ section you should make them **bold** maybe.

File: default/README.txt [hunk 1]
```diff
@@ -1,27 +1,20 @@
-# Modding Guide (Work in Progress)
+# Resource directory (default)
+The contents of the resource directory define the assets available to FreeOrion, as well as logic for modifiable content.
+These assets determine the appearance and functionality of FreeOrion.
+While much of the content within this directory is required in any resource directory, most is either required only from inclusion in other entries or is strictly optional.
 
-## Contents of the default/ directory:
+Each sub-directory has an accompaning README file within it.  This file should give a guideline for any entries or files required by FreeOrion if a new resource directory is created.
```
Comment (Morlic-fo) [inline default/README.txt]: accompanying

File: default/README.txt [hunk 2]
```diff
@@ -1,27 +1,20 @@
...
+An alternate resource directory can be specified with the --resource-dir command line flag, or by changing the <resource-dir> node in the users config.xml.
```
Comment (Morlic-fo) [inline default/README.txt]: user's

File: default/customizations/README.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Customizations
+Contains customizations, such as those that bridge between stringtable entries and scripting definitions.
+
+* custom_sitreps.txt  -  Edit this to generate sitreps as useful reminders. The file itself has more in-depth information.
+* common_user_customizations.txt  -  Another file that control the interface. See the contents for more information.
```
Comment (Morlic-fo) [inline default/customizations/README.md]: controls

**Comment by dbenage-cx** [comment]
Rebased.
Moved `default/README.txt` to `default/README.md`, for consistency.

@adrianbroher I was previewing `default/scripting/README.md` with a program that was slightly inconsistent with GitHub.  Removed the additional spaces and will rely on GitHub to preview going forward.

Thank you and @Morlic-fo for reviews

File: default/README.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+Each sub-directory has an accompanying README file within it.
+This file should give a guideline for any entries or files required by
+FreeOrion if a new resource directory is created.
+
+An alternate resource directory can be specified with the
+--resource-dir command line flag, or by changing the resource-dir node in the
+user's config.xml.
+See http://www.freeorion.org/index.php/Config for details on the config.xml
+file.
```
Comment (LGM-Doyle) [inline default/README.md:20]: The link is broken and should be http://www.freeorion.org/index.php/Config.xml.  The dot xml is missing.

File: default/README.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+## Contents
+
+* customizations/  -  User customizations
+* data/  -  Contains data files used by FreeOrion, including graphics, fonts,
+and sound.
+* python/  -  All content processed by python scripts are located here.  This
+is largely logic for AI decisions, events that occur each turn, and creation
+of a new universe when starting a new game.
+* scripting/  -  This directory contains the definitions of content within
```
Comment (LGM-Doyle) [inline default/README.md:30]: Remove the word "largely".  The list is exhaustive.  

Although in the future the list might not be exhaustive, the sentence is stronger without "largely".

Style.

File: default/README.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+FreeOrion.
+* shaders/  -  Shader effects applied to various graphics during specific game
+events.
+* stringtables/  -  Language translations.  The only required file is
+**en.txt**, which is also a fallback for any other translations.  Any
+corrections to translations, or updates for omitted languages are highly
+welcome.
+* content_specific_parameters.txt  -  **This file and the contained entries are
```
Comment (LGM-Doyle) [inline default/README.md:37]: Remove the word "highly".  Style.

File: default/README.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+required by FreeOrion**, the entries should be modified to reflect related
+content changes.  See file for details.
+* COPYING  -  License info related to any content in this directory.
+**Required file and content.**
+* credits.xml  -  Acknowledgments of those who have made FreeOrion possible. 
+**Required file and content.**
+* empire_colors.xml  -  Color options available for empires to select from. 
+**This file containing least one GG::Clr node is required.**
+* global_settings.txt  -  Contains entries which control the game mechanics.
```
Comment (LGM-Doyle) [inline default/README.md:46]: Change "containing least" -> "containing at least".  Grammar.

File: default/python/README.md [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+* common/  -  Common files for code utilized by both the AI and the server.
+* handlers/  -  see handlers/README.md
+* turn_events/  -  Python scripts that run at the beginning of every turn, and
+can trigger events that would be impossible to do purely in FOCS.
+* universe_generation/  -  Python scripts that get run at the beginning of the
+game and create the galaxy. You can customise the galaxy generation by
+editing options.py and universe_tables.py, both of which have more information
+in comments over there. The latter in specific controls which star types,
+planet types, planet sizes, and also other content get placed.
+
```
Comment (LGM-Doyle) [inline default/python/README.md:21]: Change "The latter in specific controls"  -> "The latter, specifically, controls "

**Comment by LGM-Doyle** [comment]
@dbenage-cx good work.

Ignore the style suggestions as you see fit.

**Comment by dbenage-cx** [comment]
Thank you @LGM-Doyle, rebased.

**Comment by adrianbroher** [comment]
I don't see a reason why this shouldn't be merged as-is.

> Clarification or expanded summaries of some entries may be needed

Maybe, but this isn't a pressing issue and can be done in another PR.

**Pull request merged**